### PR TITLE
Move as much configuration as possible to setup.cfg

### PR DESCRIPTION
**Pull request by astrofrog**
I need to actually check this works, including for example the package data option. Also note that this requires that setuptools be more recent than 2 years old (see https://github.com/astropy/astropy/pull/8240)

*(this is part of a series of PRs to simplify setup.py)*

### CODE REVIEW

**Comment by bsipocz** [comment]
Could you mention the setuptools version requirement change in the changelog, please?

**Comment by astrofrog** [comment]
I think https://github.com/astropy/astropy-helpers/issues/438 will help simplify things further.

File: {{ cookiecutter.package_name }}/setup.py [hunk 1]
```diff
@@ -1,155 +1,33 @@
...
-else:
-    # Get the long description from the package's docstring
-    __import__(PACKAGENAME)
-    package = sys.modules[PACKAGENAME]
-    LONG_DESCRIPTION = package.__doc__
-
 # Store the package name in a built-in variable so it's easy
 # to get from other parts of the setup infrastructure
-builtins._ASTROPY_PACKAGE_NAME_ = PACKAGENAME
+builtins._ASTROPY_PACKAGE_NAME_ = read_configuration('setup.cfg')['metadata']['name']
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/setup.py]: ooi, where is this used?

Comment (Cadair) [reply {{ cookiecutter.package_name }}/setup.py]: I can't find it in the code of astropy_helpers or core by searching on github fwiw.

Comment (astrofrog) [reply {{ cookiecutter.package_name }}/setup.py]: it's used in ``setup_pacakge.py`` for the package template but we could arguably move the definition of the package data to ``setup.cfg``

Comment (Cadair) [reply {{ cookiecutter.package_name }}/setup.py]: Where is it used? Can you link me?

**Comment by astrofrog** [comment]
Rebased on #385 for now. Still needs some cleanup which I'll try and do later today.

**Comment by bsipocz** [comment]
@astrofrog - dependency PRs have been merged, could you please have a look how this stands now?

File: {{ cookiecutter.package_name }}/docs/conf.py [hunk 2]
```diff
@@ -78,8 +78,8 @@
 # |version| and |release|, also used in various other places throughout the
 # built documents.
 
-__import__(setup_cfg['package_name'])
-package = sys.modules[setup_cfg['package_name']]
+__import__(setup_cfg['name'])
```
[the remark this answers is not in the document]
Comment (astrofrog) [reply {{ cookiecutter.package_name }}/docs/conf.py]: Fixed

File: {{ cookiecutter.package_name }}/setup.py (hunk 1 above)
Comment (astrofrog) [reply {{ cookiecutter.package_name }}/setup.py]: It's no longer in the template but it's used in a loooot of packages:

https://github.com/search?q=_ASTROPY_PACKAGE_NAME_&type=Code

I guess there's no reason we can't remove it though. Maybe we can even do batch PRs to update all the packages to hard-code the package name instead.

**Comment by astrofrog** [comment]
@bsipocz @Cadair - rebased, and I *think* good to go, but let's see what Travis says. Feel free to review again and optionally take this for a spin, or we could just merge and then do follow up PRs if needed.

File: {{ cookiecutter.package_name }}/setup.py (hunk 1 above)
Comment (astrofrog) [reply {{ cookiecutter.package_name }}/setup.py]: For now I'm leaving this in though.

File: {{ cookiecutter.package_name }}/setup.py [hunk 3]
```diff
@@ -1,147 +1,33 @@
...
 
-PACKAGENAME = metadata.get('package_name', '{{ cookiecutter.module_name }}')
-DESCRIPTION = metadata.get('description', '{{ cookiecutter.short_description|escape }}')
-AUTHOR = metadata.get('author', '{{ cookiecutter.author_name|escape }}')
-AUTHOR_EMAIL = metadata.get('author_email', '')
-LICENSE = metadata.get('license', 'unknown')
-URL = metadata.get('url', '{{ cookiecutter.project_url }}')
-__minimum_python_version__ = metadata.get("minimum_python_version", "{{ cookiecutter.minimum_python_version }}")
+# Ensure that astropy-helpers is available
+import ah_bootstrap  # noqa
```
Comment (bsipocz) [inline {{ cookiecutter.package_name }}/setup.py]: Isn't using `astropy-helpers` optional?

File: {{ cookiecutter.package_name }}/setup.py (hunk 1 above)
Comment (bsipocz) [reply {{ cookiecutter.package_name }}/setup.py]: >  Maybe we can even do batch PRs to update all the packages to hard-code the package name instead.

👍 to that

Comment (bsipocz) [reply {{ cookiecutter.package_name }}/setup.py]: (but I think we may even want to batchPR the uncontroversial things from the rendered version after we tag a new version here.)

File: {{ cookiecutter.package_name }}/setup.py (hunk 3 above)
Comment (astrofrog) [reply {{ cookiecutter.package_name }}/setup.py]: I don't think so? At least not until the APE about it is accepted. What's optional is the initialization of the submodule, right?

Comment (bsipocz) [reply {{ cookiecutter.package_name }}/setup.py]: Oh, OK then. But then it's confusing that setting the ah version is only done if it's initialized. ah_bootstrap is also versioned...

The docs is not at all helpful for the template, and I didn't check the source code.

**Comment by pllim** [comment]
> this contains temporary commits and should not be merged!

Is this still true? Seems to contradict Slack message that this is ready. Just making sure.

**Comment by astrofrog** [comment]
I’ve removed the temp commits!

**Comment by bsipocz** [comment]
OK, let's go ahead and merge this now. Thanks @astrofrog!

**Pull request merged**